File: c#/classes/ClassStaticFunction.cs
class Token {
  public static int Number;
  public string Color;

  public void Update(int number, string color) {
    Number = number;
    Color = color;
  }

  public static void Reset() {
    Number = 0;
  }
}

class ClassStaticFunction {
  public static void Main(string[] args) {
    Token token_1 = new Token();
    Token token_2 = new Token();
    Token token_3 = new Token();

    token_1.Update(1, "blue");
    token_2.Update(2, "orange");
    token_3.Update(3, "violet");

    System.Console.WriteLine("token_1.Color: " + token_1.Color);
    System.Console.WriteLine("token_2.Color: " + token_2.Color);
    System.Console.WriteLine("token_3.Color: " + token_3.Color);

    System.Console.WriteLine("\nToken.Number: " + Token.Number + "\n");

    Token.Reset();

    System.Console.WriteLine("token_1.Color: " + token_1.Color);
    System.Console.WriteLine("token_2.Color: " + token_2.Color);
    System.Console.WriteLine("token_3.Color: " + token_3.Color);

    System.Console.WriteLine("\nToken.Number: " + Token.Number + "\n");
  }
}

File: python/classes/class_static_function.py
#!/usr/bin/python3

class Token:
    number = 0

    def __init__(self):
        Token.number = 0
        self.color = ""

    def update(self, number, color):
        Token.number = number
        self.color = color

    def reset():
        Token.number = 0

token_1 = Token()
token_2 = Token()
token_3 = Token()

token_1.update(1, "blue")
token_2.update(2, "orange")
token_3.update(3, "violet")

print(f"token_1.number: {token_1.number}, token_1.color: {token_1.color}")
print(f"token_2.number: {token_2.number}, token_2.color: {token_2.color}")
print(f"token_3.number: {token_3.number}, token_3.color: {token_3.color}")

print(f"\nToken.number: {Token.number}\n")

Token.reset()

print(f"token_1.number: {token_1.number}, token_1.color: {token_1.color}")
print(f"token_2.number: {token_2.number}, token_2.color: {token_2.color}")
print(f"token_3.number: {token_3.number}, token_3.color: {token_3.color}")

print(f"\nToken.number: {Token.number}\n")

token_1.update(5, "magenta")
token_2.update(6, "amaranthine")
token_3.update(7, "navy")

print(f"token_1.number: {token_1.number}, token_1.color: {token_1.color}")
print(f"token_2.number: {token_2.number}, token_2.color: {token_2.color}")
print(f"token_3.number: {token_3.number}, token_3.color: {token_3.color}")

print(f"\nToken.number: {Token.number}\n")
